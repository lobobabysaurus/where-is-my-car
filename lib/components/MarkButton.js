import PropTypes from 'prop-types'
import React from 'react'
import { TouchableHighlight, Text } from 'react-native'

import styles from '../styles'

import GreenBlueGradient from './GreenBlueGradient'

const MarkButton = ({onPress}) =>
  <GreenBlueGradient style={styles.buttonContainer}>
    <TouchableHighlight style={styles.button} onPress={onPress}>
      <Text style={styles.text}>Mark</Text>
    </TouchableHighlight>
  </GreenBlueGradient>

MarkButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export default MarkButton
