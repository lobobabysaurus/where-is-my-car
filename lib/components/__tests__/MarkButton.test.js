import React from 'react'
import renderer from 'react-test-renderer'

import MarkButton from '../MarkButton'

describe('MarkButton', () => {
  test('renders correctly', () => {
    const component = renderer.create(
      <MarkButton onPress={jest.fn()} />
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
