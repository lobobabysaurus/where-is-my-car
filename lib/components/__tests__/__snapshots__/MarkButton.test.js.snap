// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkButton renders correctly 1`] = `
<ExponentLinearGradient
  colors={
    Array [
      4283484280,
      4278190219,
    ]
  }
  style={
    Object {
      "flexBasis": "10%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 36,
        }
      }
    >
      Mark
    </Text>
  </View>
</ExponentLinearGradient>
`;
